diff --git a/types/jsx.d.ts b/types/jsx.d.ts
index c1a509ec308613ddf03ccad5b09bdf851126432d..b1b48e95f7587b6fbdf111e718d24127b0a391ce 100644
--- a/types/jsx.d.ts
+++ b/types/jsx.d.ts
@@ -2137,5 +2137,20 @@ export namespace JSX {
     use: UseSVGAttributes<SVGUseElement>;
     view: ViewSVGAttributes<SVGViewElement>;
   }
-  interface IntrinsicElements extends HTMLElementTags, HTMLElementDeprecatedTags, SVGElementTags {}
+
+  type _IntrinsicElements = JSX.HTMLElementTags &
+    JSX.HTMLElementDeprecatedTags &
+    JSX.SVGElementTags;
+
+  type __IntrinsicElements = {
+    [Tag in keyof _IntrinsicElements]: {
+      [Attr in keyof _IntrinsicElements[Tag]]:
+        | _IntrinsicElements[Tag][Attr]
+        | null;
+    };
+  };
+
+  export interface IntrinsicElements extends __IntrinsicElements {}
+
+  // interface IntrinsicElements extends HTMLElementTags, HTMLElementDeprecatedTags, SVGElementTags {}
 }